src/Middleware/ResponseCompression/src/ResponseCompressionOptions.cs

```diff
@@ -22,8 +23,11 @@ public class ResponseCompressionOptions
 
         /// <summary>
         /// Indicates if responses over HTTPS connections should be compressed. The default is 'false'.
-        /// Enabling compression on HTTPS connections may expose security problems.
+        /// Enabling compression on HTTPS requests for remotely manipulable content may expose security problems.
         /// </summary>
+        /// <remarks>
+        /// This can be overridden per request using <see cref="IHttpsCompressionFeature"/>.
+        /// </remarks>
         public bool EnableForHttps { get; set; } = false;
```

Comment:
Is the idea to allow users to configure this to on by default (amid the potential unsafe consequences)? I'm interested on this for some Components/Blazor scenarios

Reply:
That's not the recommendation for this old setting, but it's always been possible. The new setting lets individual components opt in by default.

Reply:
Thoughts on changing this to HttpsCompressionMode and making a breaking change here?